lib/uksglist/include/uk/sglist.h

```diff
@@ -68,6 +68,32 @@ struct uk_sglist {
 	uint16_t    sg_maxseg; /* Maximum number of segment in the sg list */
 };
 
+/*
+ * Convenience macros to save the state of an sglist so it can be restored
+ * if an append attempt fails.  Since sglist's only grow we only need to
+ * save the current count of segments and the length of the ending segment.
+ * Earlier segments will not be changed by an append, and the only change
+ * that can occur to the ending segment is that it can be extended.
+ */
+struct sgsave {
```

Comment:
can you namespace this declaration since you export it?
`sglist` -> `uk_sglist`.